### ISSUE TITLE
Store collation result to disk/database

### ISSUE THREAD

**Issue by raar1** [UCL-CCS/EasyVVUQ#95]
Currently the collation element aggregates results (gradually if needed) from the decoder into a single pandas dataframe. This is important as it means all analysis elements have a single input datatype they need to handle.

However, when the campaign state is saved and reloaded, that collated dataframe is lost, as nothing is written to disk. Obviously we need to have the contents be stored, but the question is **when** and **how**.

When:
1. Upon user demand
2. Whenever campaign is saved
3. Automatically every time ```.collate()``` is called

How:
1. Store in memory as a dataframe, and write this to database upon ```store```
2. Store in memory as a dataframe, and write this to a csv file (or equivalent) upon ```store```, saving the path to the csv file in the database
3. Always store in database, and return a pandas dataframe upon demand/when needed

I'm thinking that When-3 is obviously the right choice, and I'm strongly leaning towards How-3. Any thoughts?

**Comment by raar1** [comment]
@orbitfold any ideas on this? Best practices w.r.t. to the database?

**Comment by raar1** [comment]
As @dww100 discovered, pandas dataframes already have a ```to_sql()``` method, although it is apparently notoriously slow. I'm also thinking about the best way to "update" the dataframe when we want to add new results to it.

**Comment by raar1** [comment]
Merging in the changes I've made in PR:
"Update to new collation model (see issue #95). Now there is a collater per campaign (not per app), and the data is stored in the campaign db. Multiple collates simply append to the existing db collation result. I also fleshed out the run status system a bit more ('new'->'encoded'->'collated'), and changed all run statuses in one go (hopefully more efficient). Also, collaters are now responsible for the storage/retrieval of their own collation. At present there is only one implementation which happens to use the campaign db, but in future there could be ones which use csv files, or combine multiple sub-collations etc., so this takes the load of the campaign. Lots of other small changes. All tests updated, and both the sql and json databases work with the new version."
